### ISSUE TITLE
readers v2: include previous tombstone in range tombstone changes

### ISSUE THREAD

**Issue by kbr-** [scylladb/scylla#9236]
Currently range tombstone changes contain only the new tombstone.
But if we want to efficiently reverse v2 mutation fragment streams, we'll need to efficiently find the previous tombstone for a given range tombstone change (because the previous tombstone becomes the current tombstone in the reversed stream).
Including the previous tombstone in the rtc would make this easy.

cc @denesb @avikivity @tgrabiec

**Comment by avikivity** [comment]
We don't want to reverse streams. We want to have the stream sources generate reversed streams and the intermediate readers to consume and produce reversed streams.

The data sources (cache, memtable, sstables) need to be able to produce streams in either direction, but the stream itself doesn't need to be reversed once we decided on a direction.

**Comment by denesb** [comment]
I think it is enough if our in-memory format stores the two tombstones, so the range tombstone change chain can be walked either way.

**Comment by bhalevy** [comment]
> I think it is enough if our in-memory format stores the two tombstones, so the range tombstone change chain can be walked either way.

@denesb but when we're consuming the partition in reverse, do we have the previous tombstone on the closing change?
Wouldn't the end marker hold a null tombstone (local_deletion_time) in deletion_time?

**Comment by tgrabiec** [comment]
I agree with @avikivity. Kamil, what's the use case for reversing the stream?

**Comment by kbr-scylla** [comment]
I don't remember, perhaps we were reversing the streams at some point (this issue was created when we were doing the efficient-reversed-queries for Discord effort), but now we don't have to do it (as Avi explained).

Perhaps it was the case that one stream e.g. from sstable was already reversed, but another e.g. memtable was still providing only the forward direction, so in order to merge it with the sstable stream we had to reverse it first. Maybe.

I'm closing the issue, if somebody remembers why (if) this is needed we can reopen.
